### PR TITLE
Go 1.17, Make all Tendermint and Cosmos Versions match

### PR DESCRIPTION
**Pull request by faddat**
## Summary of changes

Cosmos SDK 0.44.3
Tendermint 0.34.14
IBC-Go 1.2.2

This seems to improve relayer performance using both Hermes and Go relayer.

Replaces #589 and does the same thing more cleanly.

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [ ] Wrote tests (N/A. but let me know if any are needed)
- [ ] Updated API documentation (client/lcd/swagger-ui/swagger.yaml) <- does not change the api
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # docker build . -t cosmwasm/wasmd:latest
 # docker run --rm -it cosmwasm/wasmd:latest /bin/sh
-FROM golang:1.16-alpine3.12 AS go-builder
+FROM golang:1.17-alpine3.12 AS go-builder
```
[the remark this answers is not in the document]
Comment (faddat) [reply Dockerfile:3]: sure :)

**Review by faddat — COMMENTED** [review]
checked changed file and have modified Dockerfile.

@YunSuk-Yeo regarding ibc-go 1.2.2.... well, I agree.  I've included it here because "it works" but do not know "official guidance"

**Review by faddat — COMMENTED** [review]
viewed changed files

**Comment by faddat** [comment]
Yeah, could go with ibc-go.  We've got a wasmvm lib, at notional and I'd love your commentary on it.  It is being used for juno.  Https://github.com/notional-labs/wasmvm

**Review by faddat — COMMENTED** [review]
Yep, this works :)

**Review by faddat — COMMENTED** [review]
Basically, I realized that there were breaking changes between 1.1.* and 1.2.* but we really should use 1.1.3 otherwise we'll import multiple cosmos sdk versions and could impact stability.   There's also an older build of tendermint making its way in, and I replaced that.

File: go.mod [hunk 2]
```diff
@@ -23,9 +23,104 @@ require (
...
+	gopkg.in/ini.v1 v1.63.2 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	nhooyr.io/websocket v1.8.6 // indirect
+)
+
 replace (
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
 	github.com/cosmos/ledger-cosmos-go => github.com/terra-money/ledger-terra-go v0.11.2
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+	github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.14
```
Comment (YunSuk-Yeo) [inline go.mod:124]: I think this one is unnecessary? others seem good. 

https://github.com/cosmos/ibc-go/commit/8193efb915e6d32cd8d7b5e531c06cf32b5d2515

**Review by YunSuk-Yeo — APPROVED** [review]

**Pull request merged**